tests/unit/s2n_client_hello_test.c

```diff
@@ -91,6 +91,51 @@ int main(int argc, char **argv)
         }
     }
 
+    /* Test s2n_client_hello_get_raw_extension_by_id */
+    {
+        {
+            struct s2n_connection *conn;
+            EXPECT_NOT_NULL(conn = s2n_connection_new(S2N_SERVER));
+
```

Comment:
I actually added a new ptr free method recently, so you could avoid the s2n_connection_free at the end of your test with:
```suggestion
            DEFER_CLEANUP(struct s2n_connection *conn = s2n_connection_new(S2N_SERVER), s2n_connection_ptr_free);
            EXPECT_NOT_NULL(conn);
```

Reply:
Agreed on this update